### DISCUSSION TITLE
📣 Share Your Best GitHub Projects Tips & Innovations - Let’s create a Best Practices Cheat Sheet! ✨

### DISCUSSION THREAD

**Discussion by samus-aran** [community/community#138486] [Projects and Issues]
![Engagement Campaign Banner -](https://github.com/user-attachments/assets/3c6e36fc-5af6-4490-b3bf-e5f07df21faf)
> Alt Text: This banner image shows two text boxes with sparkles, screenshot of GitHub Projects and a Mona cat icon. Left sided box bold text: Collaborate. Share. Learn. The right sided box bold text: Share Your Best GitHub Projects Tips & Innovations. 

We’re excited to announce that we’re working on creating a **GitHub Projects Best Practices Cheat Sheet** to help users get the most out of this powerful tool! 

We’re inviting you to submit your best tips, tricks, and use cases. Whether it’s a creative workflow, a smart automation, or a unique approach to team collaboration, we’d love to hear from you about how you’re using GitHub Projects to improve productivity. 

Specifically, we’re looking for:

1. **Your workflows**: How do you set up and manage your projects?
2. **Success stories**: Can you share examples where GitHub Projects have made a big difference for you or your team?
3. **Tips and tricks**: Any shortcuts, tools, or processes that have helped you make the most of GitHub Projects?
4. **Challenges and solutions**: Have you faced any challenges in using GitHub Projects and found creative ways to overcome them?

The **best submissions** will be **collated, credited, and published** as part of a **GitHub Projects Best Practices Cheat Sheet** on Community Discussions with potential opportunities to reconnect, via the email associated with their GitHub account, with our contributors for future community collaborations.

This is your chance to showcase your expertise, inspire others, and be recognized as a contributor to a community resource!

Here’s how you can get involved:

1. **Submit your tips, stories, and use cases** by commenting on this post.
2. Include **examples** of how GitHub Projects has enhanced your task management, team collaboration, or workflow automation.
3. **Automations, customizations, or unique workflows** are especially valuable—don’t be afraid to dive into the details!

We’re excited to see how you’ve been using GitHub Projects, and we can’t wait to share these insights with the broader community. 

_Note: Responses crafted entirely by AI generation tools won't be considered as eligible. AI usage reviewing grammar and spelling are acceptable at the discretion of team members._

**Comment by htekdev** [comment]
Here is my attempt at sharing what we have done with GitHub issues, how it has helped us, and where potential opportunities exist.

We are a large company, and I oversee DevOps for the enterprise. We have established a central DevOps enablement team. With this team, we needed a scalable way to author, triage, and support issues. To streamline this, we use generative AI throughout the entire lifecycle.

## What We Did

1. **When an Issue Is Created**
Users are expected to describe what they are experiencing in a raw format with no strict guidelines on how to express their issue. We then use generative AI (with webhooks, etc.) to generate a meaningful body and title for the issue. This triggers automation to label the issue based on its type.

2. **Triaging Issues**
We employ a "Gen AI Scrum Master" that reviews issues and determines the next action items, as well as identifying the team member best suited to handle the issue. It keeps the customer up-to-date on the current status and provides our team with real-time updates on what needs to be done for each issue.

3. **Supporting Issues**
We use general automation to process comment commands for simple tasks. The cool part is that we provide the generative AI with prompts to understand which commands it can access. With simple suggestions and comments based on natural thought processes, we can close, assign, create new issues, and label them with ease. For example, if we comment something like, "This looks good, but we need @soandso to review it," our bot ensures all other assignees are removed and the issue is properly reassigned. Or if the customer says, "We are no longer experiencing this issue," the bot will automatically close the issue for us.

## Where We See Opportunities to Improve

1. **Better Integration with Existing Issue Components**
With all these different automation workflows we have, the challenge is keeping project fields up-to-date and viewing issues in customized ways. We have good templates and can automate the project field values based on various factors, but we can't trigger actions based on project fields, and the history does not show when fields were updated (except for the state).

2. **Issue Dependencies**
I see a roadmap item on this, and with the rollout of sub-issues, I'm excited about the potential. That said, having the ability to leverage relationships between issues would be great for making operations more efficient by having proper SLAs based on dependencies rather than basic priority calculations.

3. **More Generative AI!**
We have demonstrated that generative AI in the issue space is greatly beneficial and hope to see GitHub introduce and invest more in generative AI for the issue space.

4. **Reporting**
With a very proficient system for managing issues, we would like to have more control over reporting, so more customization would be helpful. Right now, we are relying on Power BI reports, which is not where I want to be.

**Reply by rantoniuk** [reply]
@htekdev The "What we did" looks already awesome! Are you using any open-sourced actions/solutions that are available on GitHub? If so, it would be awesome if you can share some links!

**Comment by xkeshav** [comment]
Best things I did for my open cource project is to create a _GitHub Community Health Files_ such as 

- CODE_OF_CONDUCT.md
- CONTRIBUTING.md
- SECURITY.md
- SUPPORT.md 

along with that customized _Issues/discussions_ templates and custom _Pull Request Template._

and adding these many files are tedious so create a npm package (and github package )  which help to generate all thes files with help of user provided details within a minute and make your repo more open source friendly.

You can check and run for your own repo

```bash
npx @xkeshav/gh-repo-care
```

see the Github Package here : https://github.com/xkeshav/gh-repo-care/pkgs/npm/gh-repo-care

**Comment by Zeroclub21** [comment]
Hello, GitHub and Azure Team!

First off, I’d like to thank you for the incredible integration of GitHub and Azure services, especially with Azure OpenAI models. It’s a powerful toolset, and I’ve enjoyed working with it. However, I wanted to share some feedback based on my recent experience in hopes that it may help improve the documentation or assist other users who might face similar challenges.

The Challenge:
While following the instructions for integrating the Azure OpenAI Chat Model using the Azure AI Inference SDK on a Windows environment, I encountered a couple of roadblocks:

ES Modules Configuration Issue: Initially, I faced an issue with the use of the import statement in my Node.js code, which led to the error:

javascript

SyntaxError: Cannot use import statement outside a module
This is because Node.js, by default, uses the CommonJS module system, which doesn't support import/export syntax unless explicitly configured. It wasn’t immediately clear from the instructions that I needed to modify the package.json to set the "type": "module".

API Key Handling Confusion (Angle Brackets): There was also some confusion around the use of angle brackets (< >) in the API key. For beginners like myself, who are still getting familiar with syntax nuances, it wasn’t immediately obvious that the key should be entered without angle brackets. Once I removed the angle brackets and passed the token as "your-github-token", everything worked perfectly.

Solutions and Suggestions:
After troubleshooting, we found two clear solutions:

ES Modules Setup: To resolve the module system issue, we simply added "type": "module" to the package.json file, which allowed Node.js to recognize and properly handle ES Modules. This small adjustment was crucial and might be worth highlighting more prominently in the documentation for those who are new to Node.js or unaware of the module system distinction.

Here’s what the modified package.json looks like:

json

{
  "name": "azure-inference-n8n",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "description": "",
  "type": "module",
  "dependencies": {
    "@azure-rest/ai-inference": "^1.0.0-beta.2",
    "@azure/core-auth": "^1.8.0",
    "@azure/core-sse": "^2.1.3"
  }
}
API Key Handling: The second issue was resolved by directly inserting the API key in the format const token = "your-github-token"; without the angle brackets. Initially, I tried to pass the key using "<key>", but this caused an "Unauthorized" error. Once the angle brackets were removed, everything worked perfectly.

Positive Outcome:
Once these adjustments were made, I was able to successfully make API calls to Azure OpenAI and retrieve results, such as having the model explain the basics of machine learning. It was a great experience to see it work in action after overcoming these small setup challenges!

Final Thoughts:
Overall, I’m really impressed with the power and potential of this integration between GitHub and Azure. With just a few tweaks to the instructions (specifically around ES Modules and API key handling), I think the setup process could be even smoother for new users.

Thanks again for the great work, and I’m excited to continue exploring more use cases with Azure OpenAI models!

Best regards,
Zeroclub21

**Comment by JorenVsyntra** [comment]
Hey everyone!

**Comment by sobujmia896** [comment]
My date of birthday  chang 13age for Father odar canssl use my al sid ok my Date of birthday SOBUJMIAH 10/07/1978 Age 46age ok world 

Uploading VID_20241129_004806.mp4…

**Comment by Andrew-Costa-10114** [comment]
Awesome!
